modules/lite/src/deckgl.js

```diff
@@ -12,13 +12,13 @@ const CANVAS_STYLE = {
   height: '100%'
 };
 
-// Add missing props for compatibility
-function normalizeProps(props) {
-  // Support old "geospatial view state as separate props" style
-  if (!props.initialViewState) {
-    const {longitude, latitude, zoom, pitch = 0, bearing = 0} = props;
-    props.initialViewState = props.viewState || {longitude, latitude, zoom, pitch, bearing};
+// Supports old "geospatial view state as separate props" style
+function getViewState(props) {
+  if (!props.viewState && 'latitude' in props && 'longitude' in props && 'zoom' in props) {
```

Comment:
Wasn't this dropped because the intention was to only support legacy top level props in the React component? 

But if those props are considered desirable in the scripting API, then I suppose that is OK. 

My main argument against it is that it does represent a small API fragmentation, and as the number of incarnations of the deck.gl API grows (react, pure-js, scripting, json, mapbox-layers etc) this having each API making its own decisions come back and bite us.

Reply:
Flat props was supported in 5.x and we did not deprecate them in 6.0.0.
I am supportive of aligning the APIs, but that would be a breaking change. This PR is intended to be a patch.